### PR TITLE
rpcclient: add more wallet commands

### PR DESCRIPTION
**Pull request by torkelrogstad**
Implement backupwallet, dumpwallet, loadwallet and unloadwallet.

### CODE REVIEW

File: btcjson/walletsvrcmds.go [hunk 1]
```diff
@@ -1030,6 +1063,9 @@ func init() {
 
 	MustRegisterCmd("addmultisigaddress", (*AddMultisigAddressCmd)(nil), flags)
 	MustRegisterCmd("addwitnessaddress", (*AddWitnessAddressCmd)(nil), flags)
+	MustRegisterCmd("loadwallet", (*LoadWalletCmd)(nil), flags)
+	MustRegisterCmd("unloadwallet", (*UnloadWalletCmd)(nil), flags)
+	MustRegisterCmd("backupwallet", (*BackupWalletCmd)(nil), flags)
```
Comment (onyb) [inline btcjson/walletsvrcmds.go:1068]: Let's sort these in alphabetic order.

File: rpcclient/wallet.go [hunk 2]
```diff
@@ -2608,12 +2608,126 @@ func (c *Client) GetWalletInfo() (*btcjson.GetWalletInfoResult, error) {
...
+// BackupWalletAsync returns an instance of a type that can be used to get the result
+// of the RPC at some future time by invoking the Receive function on the
+// returned instance.
+//
+// See BackupWallet for the blocking version and more details.
+func (c *Client) BackupWalletAsync(destination string) FutureBackupWalletResult {
+	return c.sendCmd(btcjson.NewBackupWalletCmd(destination))
+}
+
+func (c *Client) BackupWallet(destination string) error {
```
Comment (onyb) [inline rpcclient/wallet.go:2630]: Needs a comment.

**Review by onyb — CHANGES_REQUESTED** [review]
Looks good. Just needs some cosmetic changes.

**Comment by torkelrogstad** [comment]
@onyb I've addressed your comments.

**Review by onyb — APPROVED** [review]

File: rpcclient/wallet.go [hunk 3]
```diff
@@ -2608,12 +2608,128 @@ func (c *Client) GetWalletInfo() (*btcjson.GetWalletInfoResult, error) {
...
+// of the RPC at some future time by invoking the Receive function on the
+// returned instance.
+//
+// See UnloadWallet for the blocking version and more details.
+func (c *Client) UnloadWalletAsync(walletName string) FutureUnloadWalletResult {
+	return c.sendCmd(btcjson.NewUnloadWalletCmd(walletName))
+}
+
+// UnloadWallet unloads the referenced wallet
+func (c *Client) UnloadWallet(walletName string) error {
```
Comment (onyb) [inline rpcclient/wallet.go:2704]: ⚠️ `walletName` must be optional.

If the RPC server URL already contains the name of the wallet, like `http://127.0.0.1:8332/wallet/<walletname>`, it'll return an error.

From the docs:
> Specifying the wallet name on a wallet endpoint is invalid.

File: rpcclient/wallet.go [hunk 4]
```diff
@@ -2608,12 +2608,128 @@ func (c *Client) GetWalletInfo() (*btcjson.GetWalletInfoResult, error) {
...
+	return c.UnloadWalletAsync(walletName).Receive()
+}
+
+// FutureLoadWalletResult is a future promise to deliver the result of an
+// LoadWalletAsync RPC invocation (or an applicable error)
+type FutureLoadWalletResult chan *response
+
+// Receive waits for the response promised by the future
+func (r FutureLoadWalletResult) Receive() error {
+	_, err := receiveFuture(r)
```
Comment (onyb) [inline rpcclient/wallet.go:2714]: What if someone is interested in the result, which may have useful information like "warnings"?

File: rpcclient/wallet.go [hunk 5]
```diff
@@ -2608,12 +2608,128 @@ func (c *Client) GetWalletInfo() (*btcjson.GetWalletInfoResult, error) {
...
+	return c.BackupWalletAsync(destination).Receive()
+}
+
+// FutureDumpWalletResult is a future promise to deliver the result of an
+// DumpWallet RPC invocation (or an applicable error)
+type FutureDumpWalletResult chan *response
+
+// Receive waits for the response promised by the future
+func (r FutureDumpWalletResult) Receive() error {
+	_, err := receiveFuture(r)
```
Comment (onyb) [inline rpcclient/wallet.go:2642]: The result may not contain the same `destination` as the input, so we should not ignore it.

**Review by onyb — CHANGES_REQUESTED** [review]
My previous review was not so thorough, so here's a second pass.

**Comment by jcvernaleo** [comment]
Agreed with @onyb's comments then this one will be good to go in.

**Comment by torkelrogstad** [comment]
@onyb @jcvernaleo I've addressed your comments!

**Review by onyb — APPROVED** [review]

**Review by jcvernaleo — APPROVED** [review]
Looks all good now!
OK

**Pull request merged**